Client/Frontend/Browser/Playlist/PlaylistViewController.swift

```diff
@@ -0,0 +1,846 @@
+        }
+    }
+    
+    // MARK: Actions
+    
+    @objc
+    private func onExit(_ button: UIBarButtonItem) {
+        self.dismiss(animated: true, completion: nil)
+    }
+    
+    override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
+        super.viewWillTransition(to: size, with: coordinator)
+        
+        // If the player view is in fullscreen, we should NOT change the tableView layout on rotation.
+        if !playerView.isFullscreen {
+            if UIDevice.current.orientation.isLandscape {
+                navigationController?.setNavigationBarHidden(true, animated: true)
+                playerView.setFullscreenButtonHidden(true)
+                
+                tableView.isHidden = true
+                playerView.snp.remakeConstraints {
+                    $0.edges.equalToSuperview()
+                }
+            } else {
+                navigationController?.setNavigationBarHidden(false, animated: true)
+                playerView.setFullscreenButtonHidden(false)
+                
+                tableView.isHidden = false
+                let videoPlayerHeight = (1.0 / 3.0) * size.height
+                
+                playerView.snp.remakeConstraints {
+                    $0.top.equalTo(view.safeArea.top)
+                    $0.leading.trailing.equalToSuperview()
+                    $0.height.equalTo(videoPlayerHeight)
+                }
+            }
+        }
+    }
+    
+    override var preferredStatusBarStyle: UIStatusBarStyle {
+        return .lightContent
+    }
+}
+
+// MARK: UIAdaptivePresentationControllerDelegate
+
+extension PlaylistViewController: UIAdaptivePresentationControllerDelegate {
+    func adaptivePresentationStyle(for controller: UIPresentationController, traitCollection: UITraitCollection) -> UIModalPresentationStyle {
+        return .fullScreen
+    }
+}
+
+// MARK: UITableViewDataSource
+
+extension PlaylistViewController: UITableViewDataSource {
+    private func getRelativeDateFormat(date: Date) -> String {
+        if #available(iOS 13.0, *) {
+            let formatter = RelativeDateTimeFormatter()
+            formatter.unitsStyle = .abbreviated
+            formatter.dateTimeStyle = .numeric
+            return formatter.localizedString(fromTimeInterval: date.timeIntervalSinceNow)
+        } else {
+            fatalError("We're dropping iOS 12..")
+        }
+    }
+    
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return 1
+    }
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return PlaylistManager.shared.numberOfAssets()
+    }
+    
+    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        return Constants.tableRowHeight
+    }
+    
+    func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
+        return Constants.tableHeaderHeight
```

Comment:
`numberOfSections` default is 1 and can be removed.

all the rest: redundant returns